### ISSUE TITLE
Expected URL scheme 'http' or 'https' but no colon was found

### ISSUE THREAD

**Issue by fabiocosta88** [expo/expo#7935]
### Environment

  Expo CLI 3.18.6 environment info:
    System:
      OS: Windows 10 10.0.18363
    Binaries:
      Node: 13.3.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.4 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.13.1 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ^37.0.0 => 37.0.7
      react: ~16.9.0 => 16.9.0
      react-native: ~0.61.5 => 0.61.5

my app targets android, i developed it based on the expo and then ejected it, at the expo it was built perfectly, after ejecting the debug builds work perfectly, but the release builds do not open and the logcat log reports the following error:

### Expected Behavior

I just wanted the apk release to work just like the debug works

### Actual Behavior

```

`04-20 23:11:18.022 10873 10896 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
04-20 23:11:18.022 10873 10896 E AndroidRuntime: Process: com.radarcorona, PID: 10873
04-20 23:11:18.022 10873 10896 E AndroidRuntime: java.lang.IllegalArgumentException: Expected URL scheme 'http' or 'https' but no colon was found
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at okhttp3.HttpUrl$Builder.parse(HttpUrl.java:1333)
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at okhttp3.HttpUrl.get(HttpUrl.java:916)
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at okhttp3.Request$Builder.url(Request.java:165)
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at expo.modules.updates.loader.FileDownloader.addHeadersToManifestUrl(FileDownloader.java:190)
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at expo.modules.updates.loader.FileDownloader.downloadManifest(FileDownloader.java:79)
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at expo.modules.updates.loader.RemoteLoader.start(RemoteLoader.java:56)
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at expo.modules.updates.UpdatesController.lambda$start$0$UpdatesController(UpdatesController.java:307)
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at expo.modules.updates.-$$Lambda$UpdatesController$cVwvsx9XXcX6YuWbf7IvrVkJAiA.run(Unknown Source:6)
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
04-20 23:11:18.022 10873 10896 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:919)
04-20 23:11:18.024 10873 10894 W OpenGLRenderer: Failed to choose config with EGL_SWAP_BEHAVIOR_PRESERVED, retrying without...
04-20 23:11:18.036  2039  2882 W ActivityTaskManager:   Force finishing activity com.radarcorona/.MainActivity

```

**Comment by fabiocosta88** [comment]
it is worth mentioning that this is a project ejected from the expo

**Comment by cruzach** [comment]
Hey @fabiocosta88 , it looks like this is related to the `expo-updates` module. Could you explain how to reproduce this error? As you can see from the [readme](https://github.com/expo/expo/tree/master/packages/expo-updates#expo-updates), you have a few different option (maybe you are self hosting and using `expo export`?), so it would be good to include as much info as possible

**Comment by softwaredev221** [comment]
hi @fabiocosta88 did you solve this issue?

**Comment by fabiocosta88** [comment]
hi, @softwaredev221 after hard battles with the expo without results I decided to just decided to start a new project in react native, transfer all the code to the new project, fixed the imports and it worked perfectly, apparently there is some problem in the export expo. I recommend that you do the same, the process is not hard.

**Comment by cruzach** [comment]
We'd love to fix this if it is an issue, but would need one of you to list out the steps to reproduce it

**Comment by softwaredev221** [comment]
I created the expo project with latest version.
Then ejected from expo using "expo eject"
"expo start" or "expo publish"

"react-native bundle --platform android --dev false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res"

and build release apk with android studio.

I removed all codes calling expo update in MainApplication.java to fix this issue.

**Comment by fabiocosta88** [comment]
for testing purposes i followed the steps informed by @softwaredev221 and the problem was also solved.

**Comment by vasanthhr** [comment]
@fabiocosta88  and @softwaredev221 , thanks for your updates.
I am still facing this issue, please let me know what I need to remove in "MainApplication.java"  ?(android\app\src\main\java\com\facelivemodaka)


This is the log(exception logs, once I installed apk on mobile and open app by click it)

```
04-25 21:28:38.591 27453 27453 D SoLoader: init exiting
04-25 21:28:38.608 27453 27479 E Perf    : Fail to get file list com.facelivemodaka
04-25 21:28:38.608 27453 27479 E Perf    : getFolderSize() : Exception_1 = java.lang.NullPointerException: Attempt to get length of null array
04-25 21:28:38.608 27453 27479 E Perf    : Fail to get file list com.facelivemodaka
04-25 21:28:38.608 27453 27479 E Perf    : getFolderSize() : Exception_1 = java.lang.NullPointerException: Attempt to get length of null array
04-25 21:28:38.608 27453 27479 E Perf    : Fail to get file list oat
04-25 21:28:38.608 27453 27479 E Perf    : getFolderSize() : Exception_1 = java.lang.NullPointerException: Attempt to get length of null array
04-25 21:28:38.611 27453 27478 D NetworkSecurityConfig: No Network Security Config specified, using platform default
04-25 21:28:38.615 27453 27478 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
04-25 21:28:38.615 27453 27478 E AndroidRuntime: Process: com.facelivemodaka, PID: 27453
04-25 21:28:38.615 27453 27478 E AndroidRuntime: java.lang.IllegalArgumentException: Expected URL scheme 'http' or 'https' but no colon was found
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at okhttp3.HttpUrl$Builder.parse(Unknown Source:544)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at okhttp3.HttpUrl.get(Unknown Source:6)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at okhttp3.Request$Builder.url(Unknown Source:63)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at expo.modules.updates.loader.FileDownloader.addHeadersToManifestUrl(Unknown Source:9)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at expo.modules.updates.loader.FileDownloader.downloadManifest(Unknown Source:0)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at expo.modules.updates.loader.RemoteLoader.start(Unknown Source:24)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at expo.modules.updates.UpdatesController.lambda$start$0$UpdatesController(Unknown Source:20)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at expo.modules.updates.-$$Lambda$UpdatesController$cVwvsx9XXcX6YuWbf7IvrVkJAiA.run(Unknown Source:6)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
04-25 21:28:38.615 27453 27478 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:919)
04-25 21:28:38.616  4421  6399 D RecentTasksList: loadTasksInBackground: getRecentTasks
04-25 21:28:38.617  1311 27484 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
```

**Comment by brentvatne** [comment]
@softwaredev221 @fabiocosta88 - you would have been getting a crash if you didn't run `expo publish` beforehand, the error message in your logs would tell you that. we updated expo-cli to explicitly throw an error to prevent building now if you haven't published yet. further, you don't need to run `"react-native bundle --platform android --dev false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res"` - just do a release build with `./android/gradlew bundleRelease` or `yarn android --variant Release` or for ios `yarn ios --configuration Release`

@vasanthhr - did you run `expo publish`? can you share your project?

**Comment by vasanthhr** [comment]
@brentvatne  , Yes I ran` expo publish `and also` expo install expo-updates` , and then installed apk on mobile device nut still get the same below error.

I ran this command as well `cd android && ./gradlew clean`

In my root directory, even i ran `npx react-native run-android --variant=release ` by connecting mobile via USB, but still same error as below


Let me your email ID or how I can share you my project?

```
04-25 22:25:02.948  4421  6399 D RecentTasksList: dumpTasks# id=141, cmp=ComponentInfo{com.facelivemodaka/com.facelivemodaka.MainActivity}, uid=0, locked=false, isAppLocked=false
04-25 22:25:02.950  7697  7724 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
04-25 22:25:02.950  7697  7724 E AndroidRuntime: Process: com.facelivemodaka, PID: 7697
04-25 22:25:02.950  7697  7724 E AndroidRuntime: java.lang.IllegalArgumentException: Expected URL scheme 'http' or 'https' but no colon was found
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at okhttp3.HttpUrl$Builder.parse(Unknown Source:544)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at okhttp3.HttpUrl.get(Unknown Source:6)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at okhttp3.Request$Builder.url(Unknown Source:63)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at expo.modules.updates.loader.FileDownloader.addHeadersToManifestUrl(Unknown Source:9)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at expo.modules.updates.loader.FileDownloader.downloadManifest(Unknown Source:0)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at expo.modules.updates.loader.RemoteLoader.start(Unknown Source:24)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at expo.modules.updates.UpdatesController.lambda$start$0$UpdatesController(Unknown Source:20)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at expo.modules.updates.-$$Lambda$UpdatesController$cVwvsx9XXcX6YuWbf7IvrVkJAiA.run(Unknown Source:6)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
04-25 22:25:02.950  7697  7724 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:919)
04-25 22:25:02.952  1311  7730 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
04-25 22:25:02.952  1311  2407 W ActivityTaskManager:   Force finishing activity com.facelivemodaka/.MainActivity
04-25 22:25:02.953  1311  2407 D OpQuickReply: setQuickReplyResumed focusedApp AppWindowToken{353a18e token=Token{561bd89 ActivityRecord{b053090 u0 net.oneplus.launcher/.Launcher t3}}} pkgName net.oneplus.launcher
```

**Comment by softwaredev221** [comment]
@brentvatne , comment the code in MainApplication.java like this.

    @Override
    protected @Nullable String getJSBundleFile() {
//      if (!BuildConfig.DEBUG) {
        return super.getJSBundleFile();
//      } else {
//        return UpdatesController.getInstance().getLaunchAssetFile();
//      }
    }

    @Override
    protected @Nullable String getBundleAssetName() {
//      if (!BuildConfig.DEBUG) {
        return super.getBundleAssetName();
//      } else {
//        return UpdatesController.getInstance().getBundleAssetName();
//      }
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();
    SoLoader.init(this, /* native exopackage */ false);
    //initializeFlipper(this); // Remove this line if you don't want Flipper enabled

//    if (BuildConfig.DEBUG) {
//      UpdatesController.initialize(this);
//    }
  }

**Comment by manjotdhiman** [comment]
i faced same issue on expo bare project ,i think no solution available yet so now moving to react native project and will suggest the same if you facing this issue...

**Comment by ABdelh4di** [comment]
> hi @fabiocosta88 did you solve this issue?
> I faced the same bug, work well in expo and debug mode after expo eject.
> but not working in release mode with the same error in log cat

Hi softwaredev221, 

I opened an issue on one of your projects, can you please help? thank you

**Comment by briantical** [comment]
I believe I found the solution to this error.

1. Removing `expo-updates`.  As @fabiocosta88 mentioned, the issue raises from `expo-updates`.  Since I wasn't actually using the module, I followed the installation guide to remove the module. Here is the [guide](https://docs.expo.io/bare/installing-updates/).

2. If uninstallation is not an option I believe that defining the `EXPO_UPDATE_URL` and `EXPO_SDK_VERSION` in the `AndroidManifest.xml` might solve the issue. I believe this [guide](https://docs.expo.io/bare/updating-your-app/) defines how to get the `EXPO_UPDATE_URL`.
